plugins/chain_plugin/chain_plugin.cpp

```diff
@@ -2172,80 +2172,59 @@ read_only::get_table_rows_result read_only::get_kv_table_rows_context( const rea
 
    auto wait_time = end_time - cur_time;
+   bool unbounded = false;
    ///////////////////////////////////////////////////////////
    // point query
    ///////////////////////////////////////////////////////////
    if( point_query ) {
       const string &index_value = *p.index_value;
-      vector<char> iv;
-      convert_key(index_type, p.encode_type, index_value, iv);
-      vector<char> key;
-      key.resize(prefix.size() + iv.size());
-      memcpy(key.data(), prefix.data(), prefix.size());
-      memcpy(key.data() + prefix.size(), iv.data(), iv.size());
-
-      fc::variant row_var;
-      auto success = kv_context.kv_get(p.code.to_uint64_t(), key.data(), key.size(), value_size);
-      if( success ) {
-         vector<char> value;
-         value.resize(value_size);
-         auto return_size = kv_context.kv_get_data(offset, value.data(), value_size);
-         EOS_ASSERT(return_size == value_size, chain::contract_table_query_exception,  "point query value size mismatch: ${s1} ${s2}", ("s1", return_size)("s2", value_size));
-
-         if( !is_primary_idx) {
-            success = kv_context.kv_get(p.code.to_uint64_t(), value.data(), value.size(), value_size);
-            if (success) {
-               value.resize(value_size);
-               return_size = kv_context.kv_get_data(offset, value.data(), value_size);
-               EOS_ASSERT(return_size == value_size, chain::contract_table_query_exception, "point query value size mismatch: ${s1} ${s2}", ("s1", return_size)("s2", value_size));
-            }
-         }
-
-         fc::variant row_var;
-         if( p.json ) {
-            try {
-               row_var = abis.binary_to_variant( p.table.to_string(), value, abi_serializer::create_yield_function( wait_time ), shorten_abi_errors );
-            } catch ( fc::exception &e ) {
-               row_var = fc::variant(value);
-            }
-         } else {
-            row_var = fc::variant(value);
-         }
-         result.rows.emplace_back( std::move(row_var) );
-      }
-
-      return result;
+      p.lower_bound = p.index_value;
+      unbounded = true;
    }
 
    ///////////////////////////////////////////////////////////
    // ranged query
    ///////////////////////////////////////////////////////////
    bool has_lower_bound = p.lower_bound.has_value() && !p.lower_bound.value().empty();
    bool has_upper_bound = p.upper_bound.has_value() && !p.upper_bound.value().empty();
-   EOS_ASSERT(has_lower_bound && has_upper_bound, chain::contract_table_query_exception,  "Unknown range: ${t} ${i}", ("t", p.table)("i", p.index_name));
+   unbounded = !has_lower_bound || !has_upper_bound;
+   // reverse mode has upper_bound value, non-reverse and point query has lower bound value
+   EOS_ASSERT((p.reverse && has_upper_bound) || (!p.reverse && has_lower_bound) || (has_lower_bound && point_query), chain::contract_table_query_exception, "Unknown/Invalid range: ${t} ${i}", ("t", p.table)("i", p.index_name));
```

Comment:
Maybe you can change `(!p.reverse && has_lower_bound) || (has_lower_bound && point_query)` to `(!p.reverse || point_query) && has_lower_bound)` to match the comment and be consistent with `(p.reverse && has_upper_bound)`